detectable/src/main/java/com/synopsys/integration/detectable/detectables/conan/cli/ConanBaseCliDetectable.java

```diff
@@ -52,13 +50,22 @@ public DetectableResult applicable() {
 
     @Override
     public DetectableResult extractable() throws DetectableException {
-        Requirements requirements = new Requirements(fileFinder, environment);
-        conanExe = requirements.executable(() -> conanResolver.resolveConan(environment), "conan");
-        return requirements.result();
-    }
+        conanExe = conanResolver.resolveConan(environment);
 
-    @Override
-    public Extraction extract(ExtractionEnvironment extractionEnvironment) {
-        return conanCliExtractor.extract(environment.getDirectory(), conanExe);
+        String expectedVersion = getExpectedMajorConanVersion();
+        String actualVersion;
+
+        try {
+            actualVersion = conanCliExtractor.extractConanMajorVersion(environment.getDirectory(), conanExe);
```

Comment:
What happens here if customer did not provide a value for detect.conan.path?

Reply:
In that case at line 53 `conanResolver.resolveConan()` would have tried to find an installed executable based on the system PATH.

Now, it looks like if the latter fails to find conan this line will throw a null pointer exception which will get caught on the next line.

I will try to test this flow to see if it's worthwhile to have a null check before proceeding for improved error messaging.

Reply:
I've made the improvement. Thanks for pointing me in this direction.

Before the change:

```
2023-11-28 16:45:36 MST INFO  [main] ---                Conan 2 CLI: ATTEMPTED
2023-11-28 16:45:36 MST INFO  [main] ---                        Exception occurred: null
```

After the change:

```
2023-11-28 16:48:57 MST INFO  [main] ---                Conan 2 CLI: ATTEMPTED
2023-11-28 16:48:57 MST INFO  [main] ---                        No conan executable was found.
```